includes/Core/Authentication/Authentication.php

```diff
@@ -658,4 +673,95 @@ private function need_reauthenticate() {
 
 		return count( $required_and_granted_scopes ) < count( $required_scopes );
 	}
+
+	/**
+	 * Handles the exchange of a code and site code for client credentials from the proxy.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * phpcs:disable Squiz.Commenting.FunctionCommentThrowTag.Missing
+	 */
+	private function handle_site_code() {
+		$code       = $this->context->input()->filter( INPUT_GET, 'googlesitekit_code' );
+		$site_code  = $this->context->input()->filter( INPUT_GET, 'googlesitekit_site_code' );
+		$site_nonce = $this->context->input()->filter( INPUT_GET, 'googlesitekit_site_nonce' );
+
+		if ( ! $code || ! $site_code || ! $site_nonce ) {
```

Comment:
Let's not include the nonce here. If no nonce is given but the other parameters (indicating that this should be handled) are, this request should fail (similar to how it is done with the `verification_nonce`).

Reply:
You mean as `wp_die( "Invalid nonce." )` ?

Reply:
Yes, if this request is identified as "our" request (via the other two parameters), if there is no nonce, it should just die with `Invalid nonce.`.